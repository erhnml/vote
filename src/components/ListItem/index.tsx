import React from 'react';

export interface ListItemProps  { 
  post: {
    id: number,
    name: string,
    link: string,
    vote: number
  }
}

const ListItem = ({post}: ListItemProps) => {
  const {id, name, link, vote} = post;
  return (
    <p>
      {name}
    </p>
  );
}

export default ListItem;
