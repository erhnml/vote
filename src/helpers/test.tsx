import React from 'react';
import Theme from '../theme';
import { mount } from 'enzyme';

export function mountWithTheme(child: React.ReactElement) {
  return mount(child, {
      wrappingComponent: ({ children }) => <Theme>{children}</Theme>,
  });
}
