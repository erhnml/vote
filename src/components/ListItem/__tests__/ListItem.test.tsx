
import React from 'react';
import { ToastProvider } from 'react-toast-notifications';
import {  MemoryRouter } from "react-router-dom";

import { mountWithTheme } from '../../../helpers/test';
import ListItem, { DeleteIcon } from '../index';

const post = {
  id: 1,
  name: "The future of Unreal Tournament begings today",
  link: "https://unrealengine.com",
  vote: 0,
  lastVoteTime: 0
}

function ListItemWrapper({ children }:{children: React.ReactNode}) {
  return (
    <ToastProvider>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </ToastProvider>
  )
}

describe('Button', () => {
  it('renders correctly', () => {
    expect(
      <ListItemWrapper>
        <ListItem post={post} />
      </ListItemWrapper>
    ).toMatchSnapshot();
  });

  it('mount correctly', () => {
    expect(() => mountWithTheme(
      <ListItemWrapper>
        <ListItem post={post} />
      </ListItemWrapper>
    )).not.toThrow();
  });

  it('should trigger remove event correctly', () => {
    const mockFn = jest.fn();
    const wrapper = mountWithTheme(
      <ListItemWrapper>
        <ListItem post={post} onRemove={mockFn} />
      </ListItemWrapper>
    );
    wrapper.simulate('mouseover');  
    wrapper.find(DeleteIcon).simulate('click');
    expect(mockFn).toBeCalled();
  });

})
