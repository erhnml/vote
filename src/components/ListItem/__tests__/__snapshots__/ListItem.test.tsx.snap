// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<ListItemWrapper>
  <ListItem
    post={
      Object {
        "id": 1,
        "lastVoteTime": 0,
        "link": "https://unrealengine.com",
        "name": "The future of Unreal Tournament begings today",
        "vote": 0,
      }
    }
  />
</ListItemWrapper>
`;
